File: csharp/GardenCatalogue-TestDesign-Kata/GardenCatalogueTest/GardenManagementTests.cs
using GardenCatalogue;

namespace GardenCatalogueTest;

public class GardenManagementTests
{
    [Test]
    public void ManagingCatalogueScenario()
    {
        var manager = new GardenCatalogueManager(TestData.DefaultPlants);
        
        Assert.That(manager.GetPlantCount(), Is.EqualTo(9));
        
        manager.AddPlant(new PlantBuilder().WithName("Tulip").WithLatinName("Tulipa").WithArticleNumber("FLO-004").WithBloomPeriod(BloomPeriod.FromRange(Month.April, Month.May)).WithMaxHeight(0.4).WithSoil(SoilCondition.Loamy).WithLight(LightCondition.FullSun));
        Assert.That(manager.GetPlantCount(), Is.EqualTo(10));
        Assert.That(manager.GetPlantByName("Tulip"), Is.Not.Null);
        
        var springBed = manager.PlanBed(Month.April, 0.5).ToList();
        Assert.That(springBed, Has.Count.EqualTo(1));
        Assert.That(springBed[0].Name, Is.EqualTo("Tulip"));
        
        var privacyHedge = manager.PlanHedge(1.5).ToList();
        
        Assert.That(privacyHedge, Has.Count.EqualTo(5)); 
        Assert.That(privacyHedge.Any(p => p.Name == "Rose"), Is.True);
        Assert.That(privacyHedge.Any(p => p.Name == "Beech"), Is.True);
        
        var shadePlants = manager.GetPlantsForCondition(SoilCondition.Loamy, LightCondition.FullShade).ToList();
        Assert.That(shadePlants, Has.Count.EqualTo(1));
        Assert.That(shadePlants[0].Name, Is.EqualTo("Fern"));
    }
}

File: java/src/test/java/org/sammancoaching/GardenManagementTests.java
package org.sammancoaching;

import org.junit.jupiter.api.Test;
import java.util.List;
import static org.junit.jupiter.api.Assertions.*;

public class GardenManagementTests {

    @Test
    public void ManagingCatalogueScenario() {
        GardenCatalogueManager manager = new GardenCatalogueManager(TestData.defaultPlants());

        assertEquals(9, manager.getPlantCount());

        manager.addPlant(new PlantBuilder().withName("Tulip").withLatinName("Tulipa").withArticleNumber("FLO-004").withBloomPeriod(BloomPeriod.fromRange(Month.April, Month.May)).withMaxHeight(0.4).withSoil(SoilCondition.Loamy).withLight(LightCondition.FullSun).build());

        assertEquals(10, manager.getPlantCount());
        assertNotNull(manager.getPlantByName("Tulip"));

        List<Plant> springBed = manager.planBed(Month.April, 0.5);

        assertEquals(1, springBed.size());
        assertEquals("Tulip", springBed.get(0).name());

        List<Plant> privacyHedge = manager.planHedge(1.5);

        assertEquals(5, privacyHedge.size());
        assertTrue(privacyHedge.stream().anyMatch(p -> p.name().equals("Rose")));
        assertTrue(privacyHedge.stream().anyMatch(p -> p.name().equals("Beech")));

        List<Plant> shadePlants = manager.getPlantsForCondition(SoilCondition.Loamy, LightCondition.FullShade);

        assertEquals(1, shadePlants.size());
        assertEquals("Fern", shadePlants.get(0).name());
    }
}
